feat: Cassette documentation (#943)
blutui/dev.blutui.com: content/docs/cassettes/getting-started.mdx

## How do Cassettes Work?

Blutui tracks changes within the **public** and **views** directories of your project. These directories contain the front-end logic of your site. When you use the `courier push` command, these changes are uploaded to a specific cassette.

<Callout>
Cassettes only track changes in the **public** and **views** directories. Any changes made outside these directories will not be included in the cassette and won't be available when you run **courier pull**.
</Callout>

The cassette to which your changes are pushed is determined by the **cassette** attribute in your `courier.json` file. By changing this attribute, you can direct your development efforts to a different cassette, allowing you to work on multiple versions of your site simultaneously.

```json tab="courier.json"
{
  "$schema": "https://dev.blutui.com/schemas/v1/schema.json",
  "cassette": "[cassette-name]",
  "handle": "[project-handle]"
}
```

> ### Managing Cassettes

Reviewer: ```suggestion
### Typical Cassette Workflow
```

You can manage your cassettes through the Blutui Agency console. Here's a typical workflow:

![Duplicate a Cassette](https://cdn.blutui.com/uploads/assets/Dev/cassettes/duplicate-cassette.png)

> **1. Duplicate a Cassette:** To create a new version of your site (e.g., a Christmas theme), you can duplicate an existing cassette from the Agency console. This creates a new, identical copy of your site's front-end logic.

Reviewer: ```suggestion
**1. Duplicate a Cassette:** To create a new version of your project (e.g., a Christmas theme), you can duplicate an existing Cassette from the Agency Console. This creates a new, identical copy of your project's front-end logic.
```

> **2. Switch Cassettes Locally:** Update the **cassette** attribute in your `courier.json` to the name of the cassette you want to develop. Now, any changes you make and push will be applied to this new version.

Reviewer: ```suggestion
**2. Switch Cassettes Locally:** Update the `cassette` attribute in your `courier.json` config file to the name of the Cassette you want to develop. Now, any changes you make and push will be applied to this new version.
```

> **3. Develop and Test:** You can now develop your new site version (e.g., add Christmas-specific styles, layouts, and content) without affecting your live website.

Reviewer: ```suggestion
**3. Develop and Test:** You can now develop your new project version (e.g., add Christmas-specific styles, layouts, and content) without affecting your live website.
```

**4. Go Live:** Once you are satisfied with the new cassette, you can make it the live version of your site. This is done in the Agency console by selecting the **Active cassette** from a dropdown list and clicking the **Update** button.

<Callout>
The change is immediate, and your new site UI will be visible to all visitors.
</Callout>

### Restrictions on the Active Cassette

> Once you set a cassette to **Active**, it is recommended that you develop any new features or changes on a **new cassette** created by duplicating the active cassette. This ensures the production version of the site is not affected by development changes.

Reviewer: ```suggestion
Once you set a Cassette to **Active**, it is recommended that you develop any new features or changes on a **new Cassette** created by duplicating the active Cassette. This ensures the production version of the project is not affected by development changes.
```

> While you can override this and develop on the active cassette directly, it is **not recommended**. When you attempt to push to an active cassette, Courier will show a warning similar to:

Reviewer: ```suggestion
While you can override this and develop on the active Cassette directly, it is **not recommended**. When you attempt to push to an active cassette, Courier will show a warning similar to:
```

```bash
The "[cassette-name]" is currently active. You can not push to an active Cassette.

If you're feeling a little adventurous you can use the "--allow-active" flag to bypass this check.
```

To bypass this restriction, you can add the `--allow-active` flag to your `courier push` command:

```bash
courier push --allow-active
```

<Callout>
This warning will not occur if your site has not yet been published. However, once your site is live, pushing directly to the active cassette means any errors will immediately affect your production site.
</Callout>

### Example using Cassettes

As an example, imagine you have a site with a banner section: one for the **main** site and another for a **Christmas** version.

Locally, in your `courier.json` file, you set which cassette you want to develop by updating the `cassette` attribute.

You can learn how to run your project and push changes here [Running a project](http://localhost:3000/docs/getting-started/running-project)

> Below is an example where each cassette has a slightly different version of the banner section.

Reviewer: ```suggestion
Below is an example where each Cassette has a slightly different version of the banner section.
```

> **Main cassette**

Reviewer: ```suggestion
**Main Cassette**
```
